File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        if(nums.size()==1 && nums[0]==k){
            return 1;
        }
        unordered_map<int,int> mp;
        int count=0;
        int sum=0;
        mp.insert({0,1});
        for(int i=0;i<nums.size();i++){
            sum = sum+nums[i];
            if(mp.find(sum-k) != mp.end()){
                count += mp[sum-k];
            }
            mp[sum]++;
        }

        return count;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution(object):
    def subarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        result = 0
        hashmap = {}
        hashmap[0] = 1
        i = 0
        preSum =0
        while(i<len(nums)):
            preSum +=nums[i]
            if(preSum-k in hashmap):
                result +=hashmap[preSum-k]
            if(preSum not in hashmap):
                hashmap[preSum] = 1
            else:
                hashmap[preSum] += 1
            i +=1
        
        return result